tests/functional/conftest.py

```diff
@@ -1,138 +1,158 @@
 from tests.scripts.pds_request import GenericPdsRequestor, PdsRecord
 import pytest
-import asyncio
-from api_test_utils.oauth_helper import OauthHelper
-from api_test_utils.apigee_api_apps import ApigeeApiDeveloperApps
-from api_test_utils.apigee_api_products import ApigeeApiProducts
+
+from .utils.apigee_api_apps import ApigeeApiDeveloperApps
+from .utils.apigee_api_products import ApigeeApiProducts
 from .config_files import config
 import random
-from time import time
 import json
 from typing import Union
 
-from .config_files.config import BASE_URL, CLIENT_ID, CLIENT_SECRET, REDIRECT_URI
 
+from pytest_nhsd_apim.identity_service import (
+    ClientCredentialsConfig,
+    ClientCredentialsAuthenticator,
+)
+
+from pytest_nhsd_apim.apigee_apis import (
+    ApigeeClient,
+    ApiProductsAPI,
+    ApigeeNonProdCredentials,
+    DeveloperAppsAPI,
+)
+
+import logging
+
+LOGGER = logging.getLogger(__name__)
+DEVELOPER_EMAIL = "apm-testing-internal-dev@nhs.net"
+
+
+@pytest.fixture()
+def client():
+    config = ApigeeNonProdCredentials()
+    return ApigeeClient(config=config)
+
+
+@pytest.fixture()
+def api_products(client):
+    return ApiProductsAPI(client=client)
+
+
+@pytest.fixture()
+def developer_apps(client):
+    return DeveloperAppsAPI(client=client)
 
-async def _set_default_rate_limit(product: ApigeeApiProducts):
+
+@pytest.fixture()
+def add_asid_to_testapp(developer_apps, nhsd_apim_test_app):
+    app = nhsd_apim_test_app()
+    LOGGER.info(f'app:{app}')
```

Comment:
Do we still want all the logs written?

Reply:
I would keep them at least until we figure out the incosistency in using the test app with an asid attribute. They're a bit noisy, but informative

Reply:
Let's add it to the refactoring ticket [SPINEDEM-1680](https://nhsd-jira.digital.nhs.uk/browse/SPINEDEM-1680)